pkg/backingstore/reconciler.go

```diff
@@ -30,30 +30,36 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
-var bsModeToPhaseMap map[string]nbv1.BackingStorePhaseInfo
+// ModeInfo holds local information for a backing store mode.
+type ModeInfo struct {
+	Phase    nbv1.BackingStorePhase
+	Severity string
+}
+
+var bsModeInfoMap map[string]ModeInfo
 
 func init() {
-	bsModeToPhaseMap = modeToPhaseMap()
+	bsModeInfoMap = modeInfoMap()
 }
 
-func modeToPhaseMap() map[string]nbv1.BackingStorePhaseInfo {
-	return map[string]nbv1.BackingStorePhaseInfo{
-		"INITIALIZING":        {nbv1.BackingStorePhaseReady, "BackingStorePhaseReady", "Backing store mode: INITALIZING"},
-		"DELETING":            {nbv1.BackingStorePhaseReady, "BackingStorePhaseReady", "Backing store mode: DELETING"},
-		"SCALING":             {nbv1.BackingStorePhaseReady, "BackingStorePhaseReady", "Backing store mode: SCALING"},
-		"MOST_NODES_ISSUES":   {nbv1.BackingStorePhaseReady, "BackingStorePhaseReady", "Backing store mode: MOST_NODES_ISSUES"},
-		"MANY_NODES_ISSUES":   {nbv1.BackingStorePhaseReady, "BackingStorePhaseReady", "Backing store mode: MANY_NODES_ISSUES"},
-		"MOST_STORAGE_ISSUES": {nbv1.BackingStorePhaseReady, "BackingStorePhaseReady", "Backing store mode: MOST_STORAGE_ISSUES"},
-		"MANY_STORAGE_ISSUES": {nbv1.BackingStorePhaseReady, "BackingStorePhaseReady", "Backing store mode: MANY_STORAGE_ISSUES"},
-		"MANY_NODES_OFFLINE":  {nbv1.BackingStorePhaseReady, "BackingStorePhaseReady", "Backing store mode: MANY_NODES_OFFLINE"},
-		"LOW_CAPACITY":        {nbv1.BackingStorePhaseReady, "BackingStorePhaseReady", "Backing store mode: LOW_CAPACITY"},
-		"OPTIMAL":             {nbv1.BackingStorePhaseReady, "BackingStorePhaseReady", "Backing store mode: OPTIMAL"},
-		"HAS_NO_NODES":        {nbv1.BackingStorePhaseRejected, "BackingStorePhaseRejected", "Backing store mode: HAS_NO_NODES"},
-		"ALL_NODES_OFFLINE":   {nbv1.BackingStorePhaseRejected, "BackingStorePhaseRejected", "Backing store mode: ALL_NODES_OFFLINE"},
-		"NO_CAPACITY":         {nbv1.BackingStorePhaseRejected, "BackingStorePhaseRejected", "Backing store mode: NO_CAPACITY"},
-		"IO_ERRORS":           {nbv1.BackingStorePhaseRejected, "BackingStorePhaseRejected", "Backing store mode: IO_ERRORS"},
-		"STORAGE_NOT_EXIST":   {nbv1.BackingStorePhaseRejected, "BackingStorePhaseRejected", "Backing store mode: STORAGE_NOT_EXIST"},
-		"AUTH_FAILED":         {nbv1.BackingStorePhaseRejected, "BackingStorePhaseRejected", "Backing store mode: AUTH_FAILED"},
+func modeInfoMap() map[string]ModeInfo {
```

Comment:
@nb-ohad Why do you use an `init()` function to call another function just to return a map when you can create the map directly on the global var - am I missing something?
```
var bsModeInfoMap = map[string]ModeInfo{
	"INITIALIZING":        {nbv1.BackingStorePhaseReady, corev1.EventTypeNormal},
	"DELETING":            {nbv1.BackingStorePhaseReady, corev1.EventTypeNormal},
	"SCALING":             {nbv1.BackingStorePhaseReady, corev1.EventTypeNormal},
	...
}
```

Reply:
@guymguym this was in the original code. I didn't change it. But you are perfectly right.